gdx-jnigen-loader/src/main/java/com/badlogic/gdx/utils/SharedLibraryLoader.java

```diff
@@ -179,7 +179,7 @@ public void load (String libraryName) {
 
 	private InputStream readFile (String path) {
 		if (nativesJar == null) {
-			InputStream input = SharedLibraryLoader.class.getResourceAsStream("/" + path);
+			InputStream input = SharedLibraryLoader.class.getClassLoader().getResourceAsStream("/" + path);

```

Comment:
```suggestion
			InputStream input = SharedLibraryLoader.class.getClassLoader().getResourceAsStream(path);
```
`Class#getResourceAsStream` removes the leading "/" before calling `ClassLoader#getResourceAsStream`
https://github.com/openjdk/jdk/blob/80b98c86b82eb9ffd5aee67a92daa31c8b649cc6/src/java.base/share/classes/java/lang/Class.java#L3378
Have you tested these changes, whether they solve the module issue?

Reply:
I apologize. I tested the class loader approach on a tiny test module but embarrassingly didn't test it with jnigen and missed the slash. With this change applied, now the library loads successfully:

```console
$ jshell --module-path gdx-jnigen-loader.jar:lua54-platform-3.5.0-natives-desktop.jar --add-module gdx.jnigen.loader
jshell> import com.badlogic.gdx.utils.SharedLibraryLoader;
jshell> new SharedLibraryLoader().load("lua54");
jshell> // /tmp/libgdx<username>/476d0a00/liblua5464.so confirmed
```

Thank you!

Reply:
No problem, thank you for testing!
Looks good to me now!